### PR TITLE
WIP feat: add ot-commissioner

### PR DESCRIPTION
**Pull request by gocarlos**
- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by gocarlos** [comment]
depends on https://github.com/conan-io/conan-center-index/pull/1313

**Comment by gocarlos** [comment]
depends on https://github.com/conan-io/conan-center-index/pull/1314

File: recipes/ot-commissioner/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libevent/2.1.11")
+        self.requires("mdns/20200331")
+        self.requires("nlohmann_json/3.7.3")
+        self.requires("mbedtls/2.16.3-gpl")
+        self.requires("fmt/6.1.2")
+        # TODO: port to CCI
+        self.requires("cose-c/20200225@gocarlos/testing")
```
Comment (gocarlos) [inline recipes/ot-commissioner/all/conanfile.py:39]: waiting for https://github.com/conan-io/conan-center-index/pull/1314

Comment (mathbunnyru) [inline recipes/ot-commissioner/all/conanfile.py:39]:
```suggestion
        self.requires("cose-c/cci.20200430")
```

Hopefully, cose-c is already in CCI, so let's try to use it.

File: recipes/ot-commissioner/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libevent/2.1.11")
+        self.requires("mdns/20200331")
+        self.requires("nlohmann_json/3.7.3")
+        self.requires("mbedtls/2.16.3-gpl")
+        self.requires("fmt/6.1.2")
```
Comment (mathbunnyru) [inline recipes/ot-commissioner/all/conanfile.py:37]:
```suggestion
        self.requires("libevent/2.1.12")
        self.requires("mdns/20200424")
        self.requires("nlohmann_json/3.9.1")
        self.requires("mbedtls/2.25.0")
        self.requires("fmt/7.1.3")
```

File: recipes/ot-commissioner/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class OtCommissionerConan(ConanFile):
+    name = "ot-commissioner"
+    description = "OpenThread Commissioner, a Thread commissioner for joining new Thread devices and managing Thread networks."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://giþthub.com/openthread/ot-commissioner"
```
Comment (mathbunnyru) [inline recipes/ot-commissioner/all/conanfile.py:11]:
```suggestion
    homepage = "https://github.com/openthread/ot-commissioner"
```

File: recipes/ot-commissioner/all/conandata.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "20200404":
+    sha256: 04a8cd1b3e79ef8f42341d50c32916806e1313bbcc76292f7f57c36ade371b18
+    url: https://github.com/gocarlos/ot-commissioner/archive/643e7e8cf803f3fd8daa09673d49badc43b4b385.zip
```
Comment (mathbunnyru) [inline recipes/ot-commissioner/all/conandata.yml:4]: Why do you use your fork?

Let's use the upstream and the tagged version.
https://github.com/openthread/ot-commissioner/releases/tag/v0.0.1

File: recipes/ot-commissioner/all/conanfile.py (hunk 1 above)
Comment (mathbunnyru) [reply recipes/ot-commissioner/all/conanfile.py:39]: The wait is over.

File: recipes/ot-commissioner/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        self.requires("mbedtls/2.16.3-gpl")
+        self.requires("fmt/6.1.2")
+        # TODO: port to CCI
+        self.requires("cose-c/20200225@gocarlos/testing")
+        self.requires("cn-cbor/1.0.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        # extracted_dir = self.name + "-" + self.version
+        extracted_dir = self.name + "-" + os.path.basename(self.conan_data["sources"][self.version]["url"]).split(".")[0]
```
Comment (mathbunnyru) [inline recipes/ot-commissioner/all/conanfile.py:45]: Maybe it's worth using globbing?

File: recipes/ot-commissioner/all/test_package/test_package.cpp [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#include <iostream>
+
+int main(){
+    std::cout << "hello world" << std::endl;
+    return 0;
+}
```
Comment (mathbunnyru) [inline recipes/ot-commissioner/all/test_package/test_package.cpp:6]:
```suggestion
}

```

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

**Comment by gocarlos** [comment]
You are free to take over this, i changed position and therefore not using this anymore...

If someone is interested pls, pick it and improve it.... i‘m not going to be able to test this anymore and therefore leaving this open for possible interested parties

**Comment by mathbunnyru** [comment]
> You are free to take over this, i changed position and therefore not using this anymore...
> 
> If someone is interested pls, pick it and improve it.... i‘m not going to be able to test this anymore and therefore leaving this open for possible interested parties

Thanks. I will take a look. If it's easy to get done, I will finish the work you've done.

**Comment by ericLemanissier** [comment]
@danimtb the `blocked` label can be removed, now that `cose-c` is on CCI

**Pull request closed without merging**